iXBRLViewerPlugin/iXBRLViewer.py

```diff
@@ -282,15 +283,37 @@ def addFact(self, f):
 
         if f.isNumeric:
             if f.unit is not None and len(f.unit.measures[0]):
-                # XXX does not support complex units
-                unit = self.nsmap.qname(f.unit.measures[0][0])
-                aspects["u"] = unit
+                # Simplify the complex unit to a single measure by taking the first encountered measure
+                measure = self.nsmap.qname(f.unit.measures[0][0])
+                aspects['m'] = measure
```

Comment:
Why a separate key for this?

It seems like we should use the [OIM format](https://www.xbrl.org/Specification/oim-common/REC-2021-10-13/oim-common-REC-2021-10-13.html#term-unit-string-representation) for units (Arelle might already have logic for this somewhere).

Reply:
The `saveLoadableOIM` plugin does [generate a unit string representation](https://github.com/Arelle/Arelle/blob/2e1a74adf20a71389661d0f86c064797ef6a4718/arelle/plugin/saveLoadableOIM.py#L333-L348) per the above spec, the main difference being conditional parentheses. That code isn't easily accessible from this plugin, and I wanted to generate the "readable" label in a similar way anyway so my latest commit recreates the same representation.

For future improvement, I created [this issue](https://github.com/Arelle/Arelle/issues/698) which will allow Arelle and it's plugins to use the same method for unit string representation.

Reply:
If Arelle doesn't expose what we need, we can copy the relevant logic into ixbrl-viewer.  I don't think we should do pretty-printing of the local name in Python and save it in the JSON.  We should save the data in a format that the JavaScript can parse and render in whatever way is useful.